marlowe/src/Language/Marlowe/Semantics.hs

```diff
@@ -337,7 +338,7 @@ data ReduceStepResult = Reduced ReduceWarning ReduceEffect State Contract
 
 
 -- | Result of 'reduceContractUntilQuiescent'
-data ReduceResult = ContractQuiescent [ReduceWarning] [Payment] State Contract
+data ReduceResult = ContractQuiescent Bool [ReduceWarning] [Payment] State Contract
```

Comment:
Oh 🤦, you are right, that would have been much better, because that would mean that both lists are empty and that `State` and `Contract` would remain the same.
But now I really cannot be bothered to modify it everywhere and fight with Isabelle again, I'll make a low priority ticket for it ([SCP-2541](https://jira.iohk.io/browse/SCP-2541))